src/HealthSystemSecondary.java

```diff
@@ -0,0 +1,63 @@
+/**
+ * Layered implementations of secondary methods for {@code HealthSystem}.
+ */
+public abstract class HealthSystemSecondary implements HealthSystem {
+
+    /*
+     * Common methods (from Object) -------------------------------------------
+     */
+
+    /*
+     * Override the toString() method to provide a useful representation of the
+     * HealthSystem, but we avoid overriding equals() or hashCode(), as this
+     * would require access to the representation.
+     */
+
+    // CHECKSTYLE: ALLOW THIS METHOD TO BE OVERRIDDEN
+    @Override
+    public String toString() {
+        StringBuilder result = new StringBuilder("HealthSystem(");
+        result.append("Current Health: ").append(this.getHealth()).append(", ");
+        result.append("Max Health: ").append(this.getMaxHealth()).append(", ");
+        result.append("Status: ").append(this.isAlive() ? "Alive" : "Dead");
+        return result.append(")").toString();
+    }
+
+    /*
+     * Other secondary methods ------------------------------------------------
+     */
+
+    // Field to store the maximum health value
+    private int maxHealth;
```

Comment:
There shouldn't be any fields or constructors here. You access the data you need through the kernels directly.